paper/paper.md

```diff
@@ -91,6 +91,17 @@ For example, migrating zooplankton or fish can be modelled with biologically act
 For example, the `GasExchange` submodule calculates the carbon dioxide and oxygen flux at the sea surface, while the `Sediments` modules calculate fluxes of carbon and oxygen at the seafloor.
 
 We currently provide a simple Nutrient-Phytoplankton-Zooplankton-Detritus (NPZD) model [@npzd], and an intermediate complexity model, LOBSTER [@lobster], we have set up a straightforward "plug and play" framework to add additional tracers such as carbonate and oxygen chemistry systems and additional forcing. 
+A key feature of this package is the easy ability to modify models or add different formulations. 
+If a user wanted to implement a different model they could use the existing ones as a template and modify only a few lines of code where the ODEs are defined as functions.
+Alternatively, by harnessing another Julia feature, users could `import` specific functions from the models and overload only them.
+For example, if a user wanted to modify the light limitation to phytoplankton growth in the NPZD model to copy that of the LOBSTER model they could pass different parameter values to the model and write:
+
+```julia
+import OceanBioME.NPZDModel: light_limitation
+
+@inline light_limitation(PAR, α, μ₀) = 1 - exp(- PAR / α)
```

Comment:
This doesn't overload `light_limitation` (there's no type annotation?)

Reply:
Should I change it to call it redefining?

Reply:
Ah no, the code should be changed so that method extension is possible! Does this code example achieve it's intended goal? I'm not quite sure what it's trying to do. What are the "other" methods associated with the name `light_limitation`?

Reply:
In terms of what to call it, this example defines a fallback for `light_limitation`. What's unclear to me is why this would achieve any purpose. The concept of modularity and extension requires _both_ introducing a novel type, and then extending methods for that specific type to achieve some different behavior than what is already implemented.

Reply:
Ah so I think I've a) not explained this example very well and b) its definitely bad practice so maybe we shouldn't encourage it.

Here I used the first simple example I thought of and was thinking that e.g. a user wants to experiment with how the formulation of the light limitation to phytoplankton growth affects the NPZD model. This is included in the NPZD model in this line:
```julia
growth = μ₀ * Q₁₀(T) * nutrient_limitation(N, kₙ) * light_limitation(PAR, α, μ₀ * Q₁₀(T)) * P
```
so if you could change just `light_limitation`. So they could write a script that does this at the start, then runs a model. I checked and this does work to replace the function. I checked and it does do this.

Perhaps we should come up with a better example. I don't think, as currently written, it would be straightforward to modify models "properly". I do have a solution to this that I've been working on, but its a lot of changes and I was going to suggest we all talk about it at some point.